### PR TITLE
Add tests which invoke hub methods with different type param

### PR DESCRIPTION
**Pull request by Xiaohongt**
Add tests which invoke hub methods:
o   Without parameters
o   With primitive type parameters
o   With complex type parameters

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@Xiaohongt__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft Open Technologies, Inc. and real humans are currently evaluating your PR.</p>

TTYL, MSOTBOT;

File: test/signalrclient-e2e-tests/hub_connection_tests.cpp [hunk 1]
```diff
@@ -181,4 +181,127 @@ TEST(hub_connection_tests, send_message_after_reconnect)
     ASSERT_FALSE(received_event->wait(200));
 
     ASSERT_EQ(*message, U("[\"Send: test\"]"));
-}
\ No newline at end of file
+}
+
+TEST(hub_connection_tests, send_message_empty_param)
+{
+    auto hub_conn = std::make_shared<signalr::hub_connection>(url + U("SignalR"));
```
Comment (moozzyk) [inline test/signalrclient-e2e-tests/hub_connection_tests.cpp]: for hub connections `SignalR` should be added automatically. Aren't you seeing this?

File: test/signalrclient-e2e-tests/hub_connection_tests.cpp [hunk 2]
```diff
@@ -181,4 +181,127 @@ TEST(hub_connection_tests, send_message_after_reconnect)
...
+        received_event->set();
+    });
+
+    hub_conn->start().then([&hub_proxy]()
+    {
+        return hub_proxy.invoke<web::json::value>(U("send"));
+
+    }).get();
+
+    ASSERT_FALSE(received_event->wait(200));
```
Comment (moozzyk) [inline test/signalrclient-e2e-tests/hub_connection_tests.cpp]: I changed all the tests to wait up to 2000 ms as we spoke before. Please use 2000 in all new commits.

File: test/signalrclient-e2e-tests/hub_connection_tests.cpp [hunk 3]
```diff
@@ -181,4 +181,127 @@ TEST(hub_connection_tests, send_message_after_reconnect)
...
+TEST(hub_connection_tests, send_message_complex_type_return)
+{
+    auto hub_conn = std::make_shared<signalr::hub_connection>(url + U("SignalR"));
+
+    auto hub_proxy = hub_conn->create_hub_proxy(U("hubConnection"));
+
+    auto test = hub_conn->start().then([&hub_proxy]()
+    {
+        web::json::value obj{};
+        web::json::value obj1;
```
Comment (moozzyk) [inline test/signalrclient-e2e-tests/hub_connection_tests.cpp]: use meaningful variable names (like person, address)

File: test/signalrclient-e2e-tests/hub_connection_tests.cpp (hunk 1 above)
Comment (Xiaohongt) [inline test/signalrclient-e2e-tests/hub_connection_tests.cpp]: I removed "signalr" now.  I think I copied it by accidently before.

File: test/signalrclient-testhost/Connections/HubConnection.cs [hunk 4]
```diff
@@ -26,5 +26,38 @@ public string ReturnMessage(string message)
         {
             return message;
         }
+
+        public void Send()
+        {
+            Clients.Caller.send("Send");
+        }
+
+        public void SendMessages(int myint, float myfloat, double mydouble, bool mybool, char mychar)
```
Comment (moozzyk) [inline test/signalrclient-testhost/Connections/HubConnection.cs]: Use a name that describes what this function does.

File: test/signalrclient-testhost/Connections/HubConnection.cs [hunk 5]
```diff
@@ -26,5 +26,38 @@ public string ReturnMessage(string message)
         {
             return message;
         }
+
+        public void Send()
```
Comment (moozzyk) [inline test/signalrclient-testhost/Connections/HubConnection.cs]: Use a name that describes what this function does (e.g. BroadcastString)

File: test/signalrclient-testhost/Connections/HubConnection.cs [hunk 6]
```diff
@@ -26,5 +26,38 @@ public string ReturnMessage(string message)
...
+        {
+            Clients.Caller.send("Send");
+        }
+
+        public void SendMessages(int myint, float myfloat, double mydouble, bool mybool, char mychar)
+        {
+            Clients.Caller.sendMessages(myint + 1, myfloat + 1, mydouble + 1, mybool, mychar);
+        }
+
+        public void ComplexType(Person p)
```
Comment (moozzyk) [inline test/signalrclient-testhost/Connections/HubConnection.cs]: Use a name that describes what this function does (e.g. InvokeWithComplexType)

**Comment by moozzyk** [comment]
:sheep::it:

File: test/signalrclient-testhost/Startup.cs [hunk 7]
```diff
@@ -14,7 +14,9 @@ public void Configuration(IAppBuilder app)
                 map.RunSignalR<RawConnection>();
             });
 
-            app.Map("/signalr", map =>
+            app.MapSignalR();
+
+            app.Map("/custom", map =>
```
Comment (moozzyk) [inline test/signalrclient-testhost/Startup.cs]: nit: can it be just: `app.Map("/custom", map => map.RunSignalR());` ?

**Pull request merged**